Add postcode validation for Slovenia

R33D3M33R opened the pull request:
Postcode in Slovenia consists of 4 numbers. First number can be in range 1-9 and the rest can also include 0.

https://en.wikipedia.org/wiki/Postal_codes_in_Slovenia

### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changelog entry

Add postcode validation for Slovenia

R33D3M33R committed:
Add slovenian postcode validation

Postcode in Slovenia consists of 4 numbers. First number ranges from 1-9 and the rest can also include 0.

https://en.wikipedia.org/wiki/Postal_codes_in_Slovenia

R33D3M33R committed:
Add validation for Slovenian postcodes

R33D3M33R committed:
Remove Superfluous Whitespace

rodrigoprimo commented on includes/class-wc-validation.php:
```diff
@@ -96,7 +96,9 @@ public static function is_postcode( $postcode, $country ) {
 				break;
-
+			case 'SI':
+				$valid = (bool) preg_match( '/^([1-9][0-9]{3})$/i', $postcode );
```
I believe the `i` modifier is not necessary as this expression matches only numbers?

R33D3M33R replied:
Hello, you are right. I overlooked this when modifying the NL case.

rodrigoprimo reviewed (commented):
Thanks for your contribution. Overall looks good to me and it works. I left just one comment with a suggestion of a change before we merge this.

R33D3M33R committed:
Remove unnecessary caseless modifier

Since there are only numbers in the postcode, the PCRE_CASELESS modifier is superfluous

rodrigoprimo reviewed (approved):
Thanks for updating the PR. Looks good now.
